source/simulator/assemblers/newton_stokes.cc

```diff
@@ -353,15 +358,21 @@ namespace aspect
                                         :
                                         1;
 
+                  // pre-compute the tensor contractions
+                  std::vector<SymmetricTensor<2,dim> > deta_deps_times_eps_times_phi(stokes_dofs_per_cell);
+                  for (unsigned int i = 0; i < stokes_dofs_per_cell; ++i)
+                    deta_deps_times_eps_times_phi[i] = (viscosity_derivative_wrt_strain_rate * scratch.grads_phi_u[i]) * strain_rate;
+
+
                   // symmetrize when the stabilization is symmetric or SPD
                   if ((velocity_block_stabilization & Newton::Parameters::Stabilization::symmetric) != Newton::Parameters::Stabilization::none)
                     {
                       for (unsigned int i=0; i<stokes_dofs_per_cell; ++i)
                         for (unsigned int j=0; j<stokes_dofs_per_cell; ++j)
                           {
-                            data.local_matrix(i,j) += ( derivative_scaling_factor * alpha * (scratch.grads_phi_u[i] * (viscosity_derivative_wrt_strain_rate * scratch.grads_phi_u[j]) * strain_rate
-                                                                                             + scratch.grads_phi_u[j] * (viscosity_derivative_wrt_strain_rate * scratch.grads_phi_u[i]) * strain_rate)
-                                                        + derivative_scaling_factor * this->get_pressure_scaling() * scratch.grads_phi_u[i] * 2.0 * viscosity_derivative_wrt_pressure * scratch.phi_p[j] * strain_rate )
+                            data.local_matrix(i,j) += ( derivative_scaling_factor * alpha * (scratch.grads_phi_u[i] * deta_deps_times_eps_times_phi[j]
+                                                                                             + scratch.grads_phi_u[j] * deta_deps_times_eps_times_phi[i])
+                                                        + derivative_scaling_factor * pressure_scaling * 2.0 * viscosity_derivative_wrt_pressure * scratch.grads_phi_u[i] * scratch.phi_p[j] * strain_rate )
```

Comment:
since it still fails, I will first try a commit where I just change the last few `this->get_pressure_scaling()` into `pressure_scaling` and reoder this line. If that works, I we can be sure the problem is related to the tensor contraction (`deta_deps_times_eps_times_phi`).

Reply:
good plan